app/routes.py

```diff
@@ -1,2 +1,289 @@
+    task.completed_at = datetime.utcnow()
+
+    db.session.commit()
+
+    slack_bot(task.title)
+
+    response_body = {
+        "task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": task.completed_at is not None
+        }
+    }
+    return jsonify(response_body), 200
+
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete(task_id):
+    task = Task.query.get_or_404(task_id)
+    task.completed_at = None
+
+    db.session.commit()
+
+    response_body = {
+        "task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": task.completed_at is not None
+        }
+    }
+    return jsonify(response_body), 200
+
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+
+    if "title" not in request_body:
+        return jsonify({"details": "Invalid data"}), 400
+
+    new_goal = Goal(title=request_body["title"])
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    response_body = {
+        "goal": {
+            "id": new_goal.goal_id,
+            "title": new_goal.title
+        }
+    }
+    return jsonify(response_body), 201
+
+
+@goals_bp.route("", methods=["GET"])
+def read_goal():
+
+    goals = Goal.query.all()
+
+    goal_responses = []
+    for goal in goals:
+        goal_responses.append({
+            "id": goal.goal_id,
+            "title": goal.title
+        })
+
+    return jsonify(goal_responses), 200
+
+
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def read_one_goal(goal_id):
+    goal = Goal.query.get_or_404(goal_id)
+
+    response_body = {
+        "goal": {
+            "id": goal.goal_id,
+            "title": goal.title
+        }
+    }
+    return jsonify(response_body), 200
+
+
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def update_one_goal(goal_id):
+    goal = Goal.query.get_or_404(goal_id)
+
+    request_body = request.get_json()
+
+    goal.title = request_body["title"]
+
+    db.session.commit()
+
+    response_body = {
+        "goal": {
+            "id": goal.goal_id,
+            "title": goal.title,
+        }
+    }
+    return jsonify(response_body), 200
+
+
+@goals_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_one_goal(goal_id):
+    goal = Goal.query.get_or_404(goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+
+    return jsonify({
+        "details": f"Goal {goal.goal_id} \"{goal.title}\" successfully deleted"
+    }), 200
+
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"])
+def post_tasks_to_goal(goal_id):
+    request_body = request.get_json()
+
+    if "task_ids" not in request_body:
+        return jsonify({"details": "Invalid data"}), 400
+
+    task_ids = request_body["task_ids"]
+    for task_id in task_ids:
+        task = Task.query.get_or_404(task_id)
```

Comment:
This will definitely abort the process of updating all of the tasks in the request if any of them are found to be invalid. As with my prior comments about `get_or_404`, make sure to look into how to customize error handling in flask so that this still generates a JSON response.